src/components/dashboards/Admin-dashboard.tsx

```diff
@@ -1,22 +1,199 @@
+    }
+  }, [status, router]);
+
+  useEffect(() => {
+    if (session?.user?.isAdmin) {
+      const fetchRequests = async (status: RequestStatus) => {
+        return fetch(`/api/resident-request?status=${status}`)
+          .then((response) => response.json())
+          .catch((error) => {
+            console.error(`Error fetching ${status} resident requests:`, error);
+            setError(`Failed to load ${status} resident requests.`);
+            return [];
+          });
+      };
+
+      Promise.all([
+        fetchRequests(RequestStatus.PENDING),
+        fetchRequests(RequestStatus.COMPLETED),
+        fetchRequests(RequestStatus.CANCELED),
+      ]).then(([pending, completed, canceled]) => {
+        setPendingRequests(pending);
+        setFilteredPending(pending); // Initial state matches full data
+        setCompletedRequests(completed);
+        setFilteredCompleted(completed);
+        setCanceledRequests(canceled);
+        setFilteredCanceled(canceled);
+        setLoading(false);
+      });
+    }
+  }, [session?.user?.isAdmin]);
+
+  const handleSearch = (query: string) => {
+    const requestQuery = query.toLowerCase();
+
+    setFilteredPending(
+      pendingRequests.filter(
+        (request) =>
+          request.user.name.toLowerCase().includes(requestQuery) ||
+          request.user.email.toLowerCase().includes(requestQuery) ||
+          request.requestedTimeSlot.description?.toLowerCase().includes(requestQuery)
+      )
+    );
+
+    setFilteredCompleted(
+      completedRequests.filter(
+        (request) =>
+          request.user.name.toLowerCase().includes(requestQuery) ||
+          request.user.email.toLowerCase().includes(requestQuery) ||
+          request.requestedTimeSlot.description?.toLowerCase().includes(requestQuery)
+      )
+    );
 
-  if (status === "unauthenticated") {
-    router.replace("/signup");
-    return null;
+    setFilteredCanceled(
+      canceledRequests.filter(
+        (request) =>
+          request.user.name.toLowerCase().includes(requestQuery) ||
+          request.user.email.toLowerCase().includes(requestQuery) ||
+          request.requestedTimeSlot.description?.toLowerCase().includes(requestQuery)
+      )
+    );
+  };
+
+  if (error) {
+    return <p className="text-red-500">{error}</p>;
+  }
+
+  if (loading) {
+    return <p className="text-gray-500">Loading Resident Requests...</p>;
   }
 
   return (
     <>
       {session?.user?.isAdmin ? (
-        <h1>Admin Dashboard</h1>
+        <div className="flex flex-wrap">
+          {/* Left Section: Dashboard Content */}
+          <div className="w-full md:w-2/3 p-6 bg-white rounded-lg shadow-md dark:bg-simmpy-gray-900">
+            <h1 className="text-3xl font-bold text-simmpy-green">Admin Dashboard</h1>
+
+            {/* Search Bar */}
+            <SearchRequests onSearch={handleSearch} />
+
+            {/* Open Resident Requests */}
+            <Collapsible title="Open Resident Requests">
```

Comment:
I mentioned this in dev chat. To make things easier for pagination, I think we should remove the Collapsible component and display the data in a list or a table chronologically, sorted by the createdAt date